code/modules/mob/new_player/preferences_setup.dm

```diff
@@ -207,748 +207,71 @@
-				clothes_s.Blend(new /icon('icons/mob/suit.dmi', "blueshield"), ICON_OVERLAY)
-				switch(backbag)
-					if(2)
-						clothes_s.Blend(new /icon('icons/mob/back.dmi', "securitypack"), ICON_OVERLAY)
-					if(3)
-						clothes_s.Blend(new /icon('icons/mob/back.dmi', "satchel-norm"), ICON_OVERLAY)
-					if(4)
-						clothes_s.Blend(new /icon('icons/mob/back.dmi', "satchel"), ICON_OVERLAY)
-			if(JUDGE)
-				clothes_s = new /icon(uniform_dmi, "really_black_suit_s")
-				clothes_s.Blend(new /icon('icons/mob/feet.dmi', "laceups"), ICON_UNDERLAY)
-				clothes_s.Blend(new /icon('icons/mob/head.dmi', "mercy_hood"), ICON_UNDERLAY)
-				clothes_s.Blend(new /icon('icons/mob/suit.dmi', "judge"), ICON_UNDERLAY)
-				switch(backbag)
-					if(2)
-						clothes_s.Blend(new /icon('icons/mob/back.dmi', "backpack"), ICON_OVERLAY)
-					if(3)
-						clothes_s.Blend(new /icon('icons/mob/back.dmi', "satchel-norm"), ICON_OVERLAY)
-					if(4)
-						clothes_s.Blend(new /icon('icons/mob/back.dmi', "satchel"), ICON_OVERLAY)
-
-	if(disabilities & NEARSIGHTED)
-		preview_icon.Blend(new /icon('icons/mob/eyes.dmi', "glasses"), ICON_OVERLAY)
-
-	// Observers get tourist outfit.
-	if(for_observer)
-		clothes_s = new /icon(uniform_dmi, "tourist_s")
-		clothes_s.Blend(new /icon('icons/mob/feet.dmi', "black"), ICON_UNDERLAY)
-		if(backbag == 2)
-			clothes_s.Blend(new /icon('icons/mob/back.dmi', "backpack"), ICON_OVERLAY)
-		else if(backbag == 3 || backbag == 4)
-			clothes_s.Blend(new /icon('icons/mob/back.dmi', "satchel"), ICON_OVERLAY)
-
-	if(underwear_s)
-		preview_icon.Blend(underwear_s, ICON_OVERLAY)
-	if(undershirt_s)
-		preview_icon.Blend(undershirt_s, ICON_OVERLAY)
-	if(socks_s)
-		preview_icon.Blend(socks_s, ICON_OVERLAY)
-	if(clothes_s)
-		preview_icon.Blend(clothes_s, ICON_OVERLAY)
-	preview_icon.Blend(face_s, ICON_OVERLAY)
-	preview_icon_front = new(preview_icon, dir = SOUTH)
-	preview_icon_side = new(preview_icon, dir = WEST)
-
-	qdel(face_s)
-	qdel(underwear_s)
-	qdel(undershirt_s)
-	qdel(socks_s)
-	qdel(clothes_s)
+				preview_icon = icon('icons/mob/robots.dmi', "robot", SOUTH)
+				preview_icon.Scale(64, 64)
+				return
+
+	// Set up the dummy for its photoshoot
+	var/mob/living/carbon/human/dummy/mannequin = new()
+	mannequin.status_flags |= GODMODE // Why not?
+	copy_to(mannequin)
+
+	// Determine what job is marked as 'High' priority, and dress them up as such.
+	var/datum/job/previewJob
+	var/highRankFlag = job_support_high | job_medsci_high | job_engsec_high | job_karma_high
+
+	if(job_support_low & SUPPORT)
+		previewJob = job_master.GetJob("Civilian")
+	else if(highRankFlag)
+		var/highDeptFlag
+		if(job_support_high)
+			highDeptFlag = SUPPORT
+		else if(job_medsci_high)
+			highDeptFlag = MEDSCI
+		else if(job_engsec_high)
+			highDeptFlag = ENGSEC
+		else if(job_karma_high)
+			highDeptFlag = KARMA
+
+		for(var/datum/job/job in job_master.occupations)
+			if(job.flag == highRankFlag && job.department_flag == highDeptFlag)

```

Comment:
Does this... actually work? Why is it formatted like this? Why not just directly use whatever the first "high" priority job in the list is?

Reply:
its actually does work.

Reply:
and apprently if i do not have highDeptflag it breaks

Reply:
I'd advocate replacing `job.flag == highRankFlag` with `job.flag & highRankFlag`, since `highRankFlag` is a bitfield, and not a direct value - while normally only a single bit of `highRankFlag` would be set, using `&` instead of `==` would better fit with the format of data that `highRankFlag` contains.

And yes, since jobs do share bits, the department flag is important here.